src/System.Buffers/src/Resources/Strings.resx

```diff
@@ -129,4 +129,7 @@
   <data name="event_BucketExhausted" xml:space="preserve">
     <value>Bucket {0}, with {2} maximum buffers, exhausted of all Buffers of size {1} in Pool {3}</value>
   </data>
+  <data name="NotAPooledBufferException" xml:space="preserve">
+    <value>The given buffer is not a Pooled buffer and therefore cannot be returned to the Pool</value>
```

Comment:
+1, e.g. "The buffer is not associated with this pool and may not be returned to it."  Also, most of the corefx projects follow a pattern of naming the resource like "ArgumentException_BufferNotFromPool" with the exception type first and then a brief description.